oqsprov/oqsprov_capabilities.c

```diff
@@ -613,29 +613,47 @@ static int oqs_group_capability(OSSL_CALLBACK *cb, void *arg)
                 OSSL_PARAM_END                                                \
         }
 
+#    if defined(OQS_ENABLE_SIG_dilithium_2)                 \
+        || defined(OQS_ENABLE_SIG_dilithium_3)              \
```

Comment:
This will not work as and when the code generator (`generate.py`) runs again: That can extend and reduce this list arbitrarily and hard-coding it as per this proposal will break the build. I see what you try to achieve but I'm afraid it needs a new generator fragment (in https://github.com/open-quantum-safe/oqs-provider/tree/main/oqs-template/oqsprov/oqsprov_capabilities.c).